### DISCUSSION TITLE
help request:  For security reasons, I need to add basic auth authentication to the prometheus plugin

### DISCUSSION THREAD

**Discussion by zhaoyang1221** [apache/apisix#7599] [Q&A] [answered]
### Description

For security reasons, I need to add basic auth authentication to the prometheus plugin. Is there any module in apisix or openresty can do that? 
Suggestions are greatly appreciated.

### Environment

- APISIX version (run `apisix version`): 2.13.2
- Operating system (run `uname -a`): 20.04.1-Ubuntu
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): openresty/1.21.4.1

**Comment by tzssangglass** [comment, marked as answer]
we can use `public-api` plugin to protect the API that is exposed by `prometheus` plugin, ref: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/prometheus.md#specifying-metrics

and we can add `basic-auth` plugin when config the `public-api` plugin. ref: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/public-api.md#basic-usage

**Reply by zhaoyang1221** [reply]
Good idea, I'll try it
